packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -184,12 +184,25 @@ class LocalFileSystem extends local_fs.LocalFileSystem {
     List<ProcessSignal> fatalSignals = Signals.defaultExitSignals,
   }) : this(signals, fatalSignals, null);
 
+  void deletePreviousTempDirs() {
```

Comment:
Hopefully this runs before `_systemTemp` is filled in below. You could throw a `StateError` if `_systemTemp` is non-null to be extra-sure.

Reply:
good catch

Reply:
done